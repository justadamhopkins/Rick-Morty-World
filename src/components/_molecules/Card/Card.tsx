import { FC } from "react";
import Image from "next/image";
import {
  StyledCardWrapper,
  StyledCardImageWrapper,
  StyledCardLinkWrapper,
  StyledLink,
} from "./styles";
import Link from "next/link";

interface Props {
  name: string;
  gender: string;
  species: string;
  characterId: number;
  image: string;
}

export const Card: FC<Props> = ({
  name,
  gender,
  species,
  image,
  characterId,
}) => {
  return (
    <StyledCardWrapper>
      <StyledCardImageWrapper>
        <Image
          src={image}
          alt={name}
          height={300}
          width={300}
          layout="responsive"
        />
      </StyledCardImageWrapper>
      <ul>
        <li>
          <span>Name:</span> {name}
        </li>
        <li>
          <span>Gender:</span>
          {gender}
        </li>
        <li>
          <span>Species: </span>
          {species}
        </li>
      </ul>
      <StyledCardLinkWrapper>
        <Link href={`/character/${characterId}`} passHref>
          <StyledLink>View Profile</StyledLink>
        </Link>
      </StyledCardLinkWrapper>
    </StyledCardWrapper>
  );
};
