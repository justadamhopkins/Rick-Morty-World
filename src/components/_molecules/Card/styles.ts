import styled from "styled-components";

export const StyledCardWrapper = styled.div`
  position: relative;
  max-width: 300px;
  width: 100%;
  background-color: var(--white);
  ul {
    padding: 10px;
    li {
      color: var(--white);
      font-size: 1.6rem;
      padding-left: 5px;
      :nth-child(1) {
        background-color: var(--darkPurps);
      }
      :nth-child(2) {
        background-color: var(--darkLilac);
      }
      :nth-child(3) {
        background-color: var(--darkGrey);
      }
    }
  }
`;

export const StyledCardImageWrapper = styled.div`
  max-height: 250px;
  width: 100%;
  overflow: hidden;
  padding-bottom: 10px;
`;

export const StyledCardLinkWrapper = styled.div`
  display: flex;
  justify-content: center;
  padding: 10px;
`;

export const StyledLink = styled.a`
  background-color: var(--darkPink);
  width: 100%;
  max-width: 200px;
  height: 50px;
  border: none;
  transition: all 500ms ease;
  font-size: 2rem;
  text-decoration: none;
  text-align: center;
  color: var(--materialBlack);
  display: flex;
  align-items: center;
  justify-content: center;
  &:hover {
    background-color: var(--darkPurps);
    color: var(--darkPink);
  }
`;
