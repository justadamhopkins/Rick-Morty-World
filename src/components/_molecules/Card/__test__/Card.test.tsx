import { render } from "@testing-library/react";
import { Card } from "../Card";

describe("Card", () => {
  const renderCard = () => {
    return render(
      <Card
        name="mockName"
        gender="mockGender"
        species="mockSpecies"
        characterId={111}
        image="mockImagePath"
      />
    );
  };
  test("should render a single card component", () => {
    const { asFragment } = renderCard();

    expect(asFragment()).toMatchSnapshot();
  });
});
