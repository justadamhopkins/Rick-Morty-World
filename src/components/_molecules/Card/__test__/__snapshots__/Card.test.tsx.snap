// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render a single card component 1`] = `
<DocumentFragment>
  <div
    class="styles__StyledCardWrapper-sc-17jwwg1-0 dMZVdf"
  >
    <div
      class="styles__StyledCardImageWrapper-sc-17jwwg1-1 dVWWhz"
    >
      <div
        style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="display: block; box-sizing: border-box; padding-top: 100%;"
        />
        <img
          alt="mockName"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
    </div>
    <ul>
      <li>
        <span>
          Name:
        </span>
         mockName
      </li>
      <li>
        <span>
          Gender:
        </span>
        mockGender
      </li>
      <li>
        <span>
          Species: 
        </span>
        mockSpecies
      </li>
    </ul>
    <div
      class="styles__StyledCardLinkWrapper-sc-17jwwg1-2 bsPihN"
    >
      <a
        class="styles__StyledLink-sc-17jwwg1-3 llTTzT"
        href="/character/111"
      >
        View Profile
      </a>
    </div>
  </div>
</DocumentFragment>
`;
